lib/ecto/repo/schema.ex

```diff
@@ -727,7 +727,15 @@ defmodule Ecto.Repo.Schema do
         {{replace_fields!(dumper, keys), [], conflict_target}, []}
 
       :replace_all ->
-        {{replace_all_fields!(:replace_all, schema, []), [], conflict_target}, []}
+        to_remove =
+          case conflict_target do
+            target when is_atom(target) or is_list(target) ->
+              List.wrap(target)
+
+            _ ->
+              []
+          end
```

Comment:
Would this be enough?

```suggestion
        to_remove = List.wrap(conflict_target)
```

Reply:
Yes. I just wanted to make it explicit to only use `conflict_target` as to_remove if it is a field/list of fields and not a fragment.

I will change it

Reply:
@josevalim Do you want me to add some comments to the code as to why we use conflict_target as to_remove or is fine?